src/sentry/taskworker/task.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+from datetime import timedelta
+from functools import update_wrapper
+from typing import TYPE_CHECKING, Any
+
+from django.utils import timezone
+
+if TYPE_CHECKING:
+    from sentry_protos.sentry.v1.taskworker_pb2 import RetryState
+
+    from sentry.taskworker.registry import TaskNamespace
+    from sentry.taskworker.retry import Retry
+
+
+class Task:
+    name: str
+    __func: Callable[..., Any]
```

Comment:
`Callable[..., Any]` is usually a mistake / smell -- this should probably be `class Task[R, **P]` and then this should be `Callable[P, R]` to be typesafe (`P` being a [`ParamSpec`](https://youtu.be/fwZoxWyMGM8))